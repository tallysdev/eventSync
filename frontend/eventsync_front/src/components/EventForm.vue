<template>
    <v-app class="d-flex flex-column h-100 w-100">
    <NavBar />
    <v-main class="flex-grow-1 pa-16 mb-10">
        <v-container fluid class="pa-6 d-flex flex-column align-center">
            <v-form class="d-flex flex-column align-center">
                <h1 class="form-title">Criar formulário</h1>
             
                <div v-for="(question, index) in questions" :key="index" class="field-size">
                    <v-card :class="['question-card', { 'invalid-card': !question.isValid }]" outlined elevation="4">
                        <v-card-title class="d-flex justify-space-between align-center">
                            <v-select
                                v-model="question.type"
                                :items="['Discursiva', 'Múltipla escolha', 'Objetiva']"
                                label="Tipo de Pergunta"
                                outlined
                                class="mt-2"
                                @change="(value: any) => handleTypeChange(index, value)"                               
                            ></v-select>
                            
                            <v-btn @click="removeQuestion(index)" icon class="ml-2">
                                <v-icon>mdi-delete</v-icon>
                            </v-btn>

                        </v-card-title>

                        <v-card-subtitle>
                            <v-textarea
                                v-model="question.text"
                                label="Texto da Pergunta"
                                outlined
                                required
                                rows="4"
                                @input="validateQuestion(index)"          
                            ></v-textarea>
                        </v-card-subtitle>
                        
                        <v-card-subtitle> 
                            <div v-if="question.type === 'Múltipla escolha' || question.type === 'Objetiva'">
                                <v-text-field
                                    v-model="question.optionInput"
                                    label="Adicionar Opção"
                                    outlined
                                    class="mt-2"
                                ></v-text-field>

                                <v-btn @click="addOption(index)" color="primary" class="mb-4">Adicionar Opção</v-btn>

                                <div v-for="(option, optIndex) in question.optionList" :key="optIndex" class="option-container d-flex align-center mt-2">
                                    <v-textarea
                                        :value="option"
                                        readonly
                                        outlined
                                        class="option-text mr-2"
                                    ></v-textarea>

                                    <v-btn @click="removeOption(index, optIndex)" icon class="ml-2">
                                        <v-icon>mdi-delete</v-icon>
                                    </v-btn>
                                </div>
                                  
                            </div>
                        </v-card-subtitle>

                        <v-card-subtitle>
                            <div class="d-flex align-center">
                                <v-btn @click="moveQuestion(index, -1)" icon :disabled="index === 0" class="move-btn">
                                  <v-icon>mdi-arrow-up</v-icon>
                                </v-btn>
                                <v-btn @click="moveQuestion(index, 1)" icon :disabled="index === questions.length - 1" class="move-btn">
                                  <v-icon>mdi-arrow-down</v-icon>
                                </v-btn>
                              </div>
                        </v-card-subtitle>
                            
                    </v-card>
                               
                </div>

                <v-btn @click="addQuestion" color="primary"  class="mt-4">Adicionar Pergunta</v-btn>
                <v-btn color="primary" class="mt-4">Salvar Formulário</v-btn>
                
            </v-form>

            <v-divider class="my-4"></v-divider>

            <h2 class="form-title">Visualizar Formulário</h2>
            <FormPreview :questions="questions" />

        </v-container>
    </v-main>
    <FooterVue />
    
    </v-app>
</template>
  
<script setup lang="ts">
import NavBar from '../components/NavBar.vue'
import FooterVue from '../components/Footer.vue'
import FormPreview from '../components/FormPreview.vue'
import type { QuestionCreate } from '@/types/types';
import { ref } from 'vue'

const questions = ref<QuestionCreate[]>([{ text: '', type: 'Discursiva', optionList: [], isValid: true}])

const addQuestion = () => {
    validateQuestions()
    questions.value.push({ text: '', type: 'Discursiva', optionList: [], isValid: true })
}

const removeQuestion = (index: number) => {
    questions.value.splice(index, 1)
}

const addOption = (index: number) => {
    const question = questions.value[index];
    if (question.optionInput?.trim()) {
        question.optionList.push(question.optionInput.trim())
        question.optionInput = ''
        validateQuestion(index);
    } else {
        alert('Digite uma opção válida')
    }
}

const moveQuestion = (index: number, direction: any) => {
    const newIndex = index + direction
    if (newIndex < 0 || newIndex >= questions.value.length) return
    const movedQuestion = questions.value.splice(index, 1)[0]
    questions.value.splice(newIndex, 0, movedQuestion)
}

const removeOption = (questionIndex: number, optionIndex: number) => {
    questions.value[questionIndex].optionList.splice(optionIndex, 1)
    validateQuestion(questionIndex)
}

const handleTypeChange = (index: number, type: 'Discursiva' | 'Múltipla escolha' | 'Objetiva') => {
    const question = questions.value[index]
    question.type = type

    if (type === 'Discursiva') {
        question.optionList = []
    } 
    validateQuestion(index)
}

const validateQuestion = (index: number) => {
    const question = questions.value[index]
    
    if(question.type === 'Discursiva') {
        question.isValid = question.text.trim() !== ''
    }
    else {
        question.isValid = question.text.trim() !== '' && question.optionList.length > 0
    }
}

const validateQuestions = () => {
    questions.value.forEach((_: any, index: number) => validateQuestion(index))
}

</script>

<style scoped>
.form-title {
    text-align: center;
    margin-bottom: 30px;
    font-size: 2.8rem;
}

.field-size {
    max-width: 450px; 
    min-width: 300px;
    width: 100%;
    margin-bottom: auto;
    font-size: 1.2rem;
    margin-left: auto;
    margin-right: auto; 
}

.question-card {
    width: 100%;
    margin-top: 16px;
}

.invalid-card {
    background-color: #fa8a93;
    border-color: #55010a;
    color: #721c24;
}

.mt-4 {
    margin-top: 16px;
}

.ml-2 {
    color: red;
}

.move-btn {
    margin-bottom: 16px;
    margin-left: 8px;
}

</style>