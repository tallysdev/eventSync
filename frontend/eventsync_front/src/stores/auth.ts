import { defineStore } from 'pinia'
import api from '@/services/api'
import { jwtDecode, type JwtPayload } from 'jwt-decode'
import {type User, type AuthState, type Profile} from '../types/users'


export const useAuthStore = defineStore({
  persist: true,
  id: 'auth',
  state: (): AuthState => ({
    user: null,
    token: null,
    profile: null,
    rtoken: null,
  }),

  getters: {
    getUser: (state): Profile | null => state.profile,
    isAuthenticated: (state): boolean => !!state.token,
    hasAdminPerm: (state): boolean => !!state.profile?.is_staff,
    hasProfile: (state): boolean => !!state.profile,
  },
  actions: {
    async login(user: User) {
      try {
        const resp = await api.post('login/', {
          email: user.email,
          password: user.password
        })
        if (resp.status === 200) {
          const access_token = resp.data.access
          const refresh_token = resp.data.refresh
          this.setToken(access_token)
          this.setRefreshToken(refresh_token)
          localStorage.setItem('token', access_token)
          localStorage.setItem('refresh_token', refresh_token)
          localStorage.setItem('email', user.email)
          await this.setProfile(resp.data.access)
        }
      } catch (err) {
        alert(err)
      }
    },
    logout() {
      this.clearTokens()
      this.clearProfile()
      localStorage.removeItem('token')
      localStorage.removeItem('refresh_token')
      localStorage.removeItem('email')
      localStorage.removeItem('profile')
    },
    checkAuth() {
      const token = localStorage.getItem('token')
      if (token && token !== this.token) {
        this.setToken(token)
      }
      if (!isSignedIn()) {
        this.logout()
      }
    },
    async setProfile(token: string) {
      type customJwtPayload = JwtPayload & { user_id: number }
      const payload = jwtDecode<customJwtPayload>(token)
      const userID: number = payload.user_id
      try {
        const resp = await api.get(`users/${userID}/`, {
          headers: {
            Authorization: `Bearer ${token}`
          }
        })
        if (resp.status === 200) {
          const profile = resp.data
          this.profile = profile
          localStorage.setItem('profile', JSON.stringify(profile))
        }
      } catch (err) {
        console.error('Erro ao obter perfil do usuário:', err)
      }
    },

    setToken(token: string) {
      this.token = token
    },

    setRefreshToken(token: string) {
      this.rtoken = token
    },
    clearTokens() {
      this.token = null
      this.rtoken = null
    },
    clearProfile() {
      this.profile = null
    }
  }
})

function isSignedIn() {
  const token = localStorage.getItem('token')

  if (!token)
    // Se não existe o token no LocalStorage
    return false // significa que o usuário não está assinado.

  try {
    const { exp: expiration } = jwtDecode<{ exp: number }>(token)
    const isExpired = !!expiration && Date.now() > expiration * 1000

    if (isExpired)
      // Se o token tem uma data de expiração e
      return false // essa data é menor que a atual o usuário
    // não está mais assinado.
    return true
  } catch (_) {
    // O "jwt-decode" lança erros pra tokens inválidos.
    return false // Com um token inválido o usuário não está assinado.
  }
}
