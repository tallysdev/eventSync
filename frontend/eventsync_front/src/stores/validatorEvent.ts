import type { FormValuesEvent } from '@/types/event'
import { ref } from 'vue'

export const snackbar = ref(false)
export const snackbarText = ref('')
export const snackbarColor = ref('')

const showSnackbar = (message: string, type: string) => {
  snackbarText.value = message
  snackbarColor.value = type === 'success' ? 'green' : 'red'
  snackbar.value = true
}

export const validateFields = (formValues: FormValuesEvent): boolean => {
  const {
    name,
    start_date,
    end_date,
    local,
    min_quantity,
    max_quantity,
    hours_quantity,
    event_type,
    description,
    status
  } = formValues

  const fields = [
    { value: name, message: 'Nome do Evento é obrigatório' },
    { value: start_date, message: 'Data de Início do Evento é obrigatória' },
    { value: end_date, message: 'Data de Fim do Evento é obrigatória' },
    { value: local, message: 'Local do Evento é obrigatório' },
    {
      value: min_quantity && min_quantity > 0,
      message: 'Quantidade Mínima de Participantes é obrigatória e deve ser maior que 0'
    },
    {
      value: max_quantity && max_quantity > 0,
      message: 'Quantidade Máxima de Participantes é obrigatória e deve ser maior que 0'
    },
    {
      value: hours_quantity && hours_quantity > 0,
      message: 'Quantidade de Horas é obrigatória e deve ser maior que 0'
    },
    { value: event_type, message: 'Tipo do Evento é obrigatório' },
    { value: description, message: 'Descrição do Evento é obrigatória' }
  ]

  for (const field of fields) {
    if (!field.value) {
      showSnackbar(field.message, 'error')
      return false
    }
  }

  const startDate = new Date(start_date)
  const endDate = new Date(end_date)
  if (startDate > endDate) {
    showSnackbar('A Data de Início do Evento deve ser anterior à Data de Fim', 'error')
    return false
  }

  return true
}

export const validateNumberInput = (event: KeyboardEvent) => {
  const key = event.key
  if (!/[0-9]/.test(key)) {
    event.preventDefault()
  }
}
