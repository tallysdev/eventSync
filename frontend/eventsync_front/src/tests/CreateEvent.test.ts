import { mount } from '@vue/test-utils'
import { describe, it, expect, beforeEach, vi } from 'vitest'
import EventForm from '@/components/events/CreateEventPage.vue'
import axios from 'axios'
import { snackbar, snackbarColor, snackbarText } from '@/stores/validatorEvent'

vi.mock('axios')

describe('CreateEventPage.vue', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = mount(EventForm)
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('contains NavBar component', () => {
    expect(wrapper.findComponent({ name: 'NavBar' }).exists()).toBe(true)
  })

  it('contains Footer component', () => {
    expect(wrapper.findComponent({ name: 'Footer' }).exists()).toBe(true)
  })

  it('contains form fields', () => {
    expect(wrapper.find('v-text-field[label="Nome do Evento"]').exists()).toBe(true)
    expect(wrapper.find('v-text-field[label="Data de Início do Evento"]').exists()).toBe(true)
    expect(wrapper.find('v-text-field[label="Data de Fim do Evento"]').exists()).toBe(true)
    expect(wrapper.find('v-text-field[label="Quantidade Mínima de Participantes"]').exists()).toBe(true)
    expect(wrapper.find('v-text-field[label="Quantidade Máxima de Participantes"]').exists()).toBe(true)
    expect(wrapper.find('v-text-field[label="Quantidade de Horas"]').exists()).toBe(true)
    expect(wrapper.find('v-textarea[label="Descrição do Evento"]').exists()).toBe(true)
    expect(wrapper.find('v-select[label="Localização do Evento"]').exists()).toBe(true)
    expect(wrapper.find('v-select[label="Tipo do Evento"]').exists()).toBe(true)
    expect(wrapper.find('v-btn[type="submit"]').text()).toContain('Criar Evento')
    expect(wrapper.find('v-snackbar').exists()).toBe(true)
  })

  it('fetches locations on mount', async () => {
    const mockData = {
      data: {
        results: [
          { id: 1, local_name: 'Local A' },
          { id: 2, local_name: 'Local B' },
        ],
      },
    }
    axios.get.mockResolvedValueOnce(mockData)

    await wrapper.vm.$nextTick()
    await wrapper.vm.fetchLocations()

    expect(wrapper.vm.locations).toEqual(mockData.data.results)
    expect(axios.get).toHaveBeenCalledWith('http://127.0.0.1:8000/eventsync/api/v1/locals')
  })

  // it('submits form and creates event successfully', async () => {
  //   const mockData = {
  //     data: {
  //       results: [
  //         { id: 1, local_name: 'Local A' },
  //         { id: 2, local_name: 'Local B' },
  //       ],
  //     },
  //   }
  //   axios.get.mockResolvedValueOnce(mockData)

  //   await wrapper.vm.$nextTick()
  //   await wrapper.vm.fetchLocations()

  //   const mockResponse = {
  //     data: {
  //       id: 1,
  //       name: 'Evento Teste',
  //       start_date: '2024-08-10',
  //       end_date: '2024-08-11',
  //       status: 'upcoming',
  //       local: 1,
  //       description: 'Descrição do evento teste',
  //       min_quantity: 10,
  //       max_quantity: 100,
  //       hours_quantity: 2,
  //       event_type: 'conference',
  //     },
  //   }
  //   axios.post.mockResolvedValueOnce(mockResponse)


  //   wrapper.setData({
  //     name: 'Evento Teste',
  //     start_date: '2024-08-10',
  //     end_date: '2024-08-11',
  //     local: 1,
  //     min_quantity: 10,
  //     max_quantity: 100,
  //     hours_quantity: 2,
  //     event_type: 'conference',
  //     description: 'Descrição do evento teste',
  //   })

  //   await wrapper.find('v-btn[type="submit"]').trigger('click')

  //   expect(axios.post).toHaveBeenCalledWith('http://127.0.0.1:8000/eventsync/api/v1/events/', {
  //     name: 'Evento Teste',
  //     start_date: '2024-08-10',
  //     end_date: '2024-08-11',
  //     status: 'upcoming',
  //     local: 1,
  //     description: 'Descrição do evento teste',
  //     min_quantity: 10,
  //     max_quantity: 100,
  //     hours_quantity: 2,
  //     event_type: 'conference',
  //   })

  //   expect(snackbar.value).toBe(true)
  //   expect(snackbarText.value).toBe('Evento criado com sucesso!')
  //   expect(snackbarColor.value).toBe('green')

  // })

  // it('handles event creation error', async () => {
  //   const errorResponse = {
  //     response: {
  //       data: {
  //         message: 'Erro ao criar evento',
  //       },
  //     },
  //   }
  //   axios.post.mockRejectedValueOnce(errorResponse)

  //   wrapper.setData({
  //     name: 'Evento Teste',
  //     start_date: '2024-08-10',
  //     end_date: '2024-08-11',
  //     local: 'Local A',
  //     min_quantity: 10,
  //     max_quantity: 100,
  //     hours_quantity: 2,
  //     event_type: 'conference',
  //     description: 'Descrição do evento teste',
  //   })

  //   await wrapper.find('v-btn[type="submit"]').trigger('click')
  //   await wrapper.vm.$nextTick()  // Aguarda a atualização do estado

  //   expect(snackbar.value).toBe(true)
  //   expect(snackbarText.value).toBe('Erro ao criar evento')
  //   expect(snackbarColor.value).toBe('red')
  // })
})

