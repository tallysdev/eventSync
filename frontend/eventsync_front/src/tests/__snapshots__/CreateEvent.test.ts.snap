// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CreateEventPage.vue > renders correctly 1`] = `
"<v-app data-v-8bf04b29="" class="d-flex flex-column w-100">
  <div data-v-8bf04b29="">
    <!-- Desktop Navbar -->
    <v-app-bar app="" color="secondary" dense="" flat="" class="py-2 mb-2 hidden-sm-and-down fill-width">
      <v-container>
        <v-row class="d-flex align-center justify-space-evenly">
          <v-col cols="1">
            <v-img src="../../favicon.ico" alt="Platform Logo" height="40"></v-img>
          </v-col>
          <v-col cols="3">
            <v-text-field modelvalue="" color="primary" bg-color="complementary" placeholder="Pesquisar eventos..." prepend-inner-icon="mdi-magnify" variant="outlined" flat="" hide-details="" rounded=""></v-text-field>
          </v-col>
          <v-col cols="3">
            <v-select items="Localização 1,Localização 2,Localização 3" item-text="name" item-value="id" label="Localizações" prepend-inner-icon="mdi-map-marker" variant="solo" density="comfortable" bg-color="transparent" flat="" rounded="" hide-details="" class="pr-6"></v-select>
          </v-col>
          <v-col cols="4" class="d-flex align-center justify-space-around">
            <v-btn to="[object Object]">Crie seu evento</v-btn>
            <v-btn>Acesse sua conta</v-btn>
            <v-btn class="text-none" color="primary" variant="flat" style="border: 1px solid white;" rounded="xs"><b>Cadastre-se</b></v-btn>
          </v-col>
        </v-row>
      </v-container>
    </v-app-bar><!-- Mobile Navbar -->
    <v-app-bar app="" color="secondary" dense="" flat="" class="py-2 mb-0 hidden-md-and-up" style="position: fixed; top: 0px; left: 0px; right: 0px;">
      <v-container>
        <v-row class="d-flex align-center justify-space-between">
          <v-col cols="2">
            <v-img src="../../favicon.ico" alt="Platform Logo" height="40"></v-img>
          </v-col>
          <v-col cols="2" class="d-flex align-center justify-end">
            <v-btn icon="">
              <v-icon>mdi-menu</v-icon>
            </v-btn>
          </v-col>
        </v-row>
      </v-container>
    </v-app-bar><!-- Mobile Navigation Dialog -->
    <v-dialog modelvalue="false" fullscreen="" hide-overlay="" transition="dialog-bottom-transition">
      <v-card>
        <v-toolbar flat="" color="secondary">
          <v-toolbar-title>Mobile Menu</v-toolbar-title>
          <v-spacer></v-spacer>
          <v-btn icon="">
            <v-icon>mdi-close</v-icon>
          </v-btn>
        </v-toolbar>
        <v-list>
          <v-list-item>
            <v-list-item-title>Pesquisar eventos</v-list-item-title>
          </v-list-item>
          <v-list-item>
            <v-list-item-title>Crie seu evento</v-list-item-title>
          </v-list-item>
          <v-list-item>
            <v-list-item-title>Acesse sua conta</v-list-item-title>
          </v-list-item>
          <v-list-item>
            <v-list-item-title>Cadastre-se</v-list-item-title>
          </v-list-item>
          <v-list-group>
            <v-list-item>
              <v-list-item-title>Localização 1</v-list-item-title>
            </v-list-item>
            <v-list-item>
              <v-list-item-title>Localização 2</v-list-item-title>
            </v-list-item>
            <v-list-item>
              <v-list-item-title>Localização 3</v-list-item-title>
            </v-list-item>
          </v-list-group>
        </v-list>
        <!--v-if-->
      </v-card>
    </v-dialog>
  </div>
  <v-main data-v-8bf04b29="" class="flex-grow-1 pa-16 mb-10">
    <v-container data-v-8bf04b29="" fluid="" class="pa-6 d-flex flex-column align-center">
      <h1 data-v-8bf04b29="" class="form-title">Crie seu Evento</h1>
      <v-form data-v-8bf04b29="" class="pa-3 d-flex flex-column align-center max-width-form">
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Nome do Evento" required="" class="field-size"></v-text-field>
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Data de Início do Evento" type="date" required="" class="field-size"></v-text-field>
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Data de Fim do Evento" type="date" required="" class="field-size"></v-text-field>
        <v-select data-v-8bf04b29="" class="field-size" label="Localização do Evento" modelvalue="" items="" item-title="local_name" item-value="id" required=""></v-select>
        <v-select data-v-8bf04b29="" modelvalue="" items="conference,workshop,seminar,meetup" label="Tipo do Evento" required="" class="field-size"></v-select>
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Quantidade Mínima de Participantes" type="number" required="" class="field-size" min="1"></v-text-field>
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Quantidade Máxima de Participantes" type="number" required="" class="field-size" min="1"></v-text-field>
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Quantidade de Horas" type="number" required="" class="field-size" min="1"></v-text-field>
        <v-textarea data-v-8bf04b29="" modelvalue="" label="Descrição do Evento" required="" class="field-size"></v-textarea>
        <v-row data-v-8bf04b29="" class="mt-4">
          <v-col data-v-8bf04b29="" cols="6" class="d-flex justify-start">
            <v-btn data-v-8bf04b29="" color="secondary" class="btn-size btn-style voltar-btn" elevation="2">Voltar</v-btn>
          </v-col>
          <v-col data-v-8bf04b29="" cols="6" class="d-flex justify-end">
            <v-btn data-v-8bf04b29="" type="submit" color="primary" class="btn-size btn-style" elevation="2">Criar Evento</v-btn>
          </v-col>
        </v-row>
      </v-form>
    </v-container>
  </v-main>
  <v-footer data-v-4896eafd="" data-v-8bf04b29="" color="secondary" class="py-6" width="auto" height="auto">
    <v-container data-v-4896eafd="">
      <v-row data-v-4896eafd="" class="footer-row" align="center" justify="space-between">
        <v-col data-v-4896eafd="" cols="12" md="2" class="d-flex align-center justify-center mb-4 mb-md-0">
          <v-img data-v-4896eafd="" src="../../favicon.ico" alt="Platform Logo" height="40"></v-img>
        </v-col>
        <v-col data-v-4896eafd="" cols="12" md="3" class="text-center text-md-left">
          <h3 data-v-4896eafd="" class="text-h6 font-weight-bold">Encontre Eventos</h3>
          <v-list data-v-4896eafd="" dense="">
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Hoje</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Amanhã</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Esta Semana</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Próxima Semana</v-list-item-title>
            </v-list-item>
          </v-list>
        </v-col>
        <v-col data-v-4896eafd="" cols="12" md="3" class="text-center text-md-left">
          <h3 data-v-4896eafd="" class="text-h6 font-weight-bold">Mapa do site</h3>
          <v-list data-v-4896eafd="" dense="">
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Home</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Eventos</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Login</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Contato</v-list-item-title>
            </v-list-item>
          </v-list>
        </v-col>
        <v-col data-v-4896eafd="" cols="12" md="4" class="text-center text-md-left">
          <h3 data-v-4896eafd="" class="text-h6 font-weight-bold">Contato</h3>
          <v-list data-v-4896eafd="" dense="">
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Email: fakeemail@gmail.com</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Celular: 213214124124</v-list-item-title>
            </v-list-item>
          </v-list>
        </v-col>
      </v-row>
    </v-container>
  </v-footer>
  <v-snackbar data-v-8bf04b29="" modelvalue="true" color="red" timeout="3000" top="">Erro ao buscar locais</v-snackbar>
</v-app>"
`;

exports[`EventForm.vue > renders correctly 1`] = `
"<v-app data-v-8bf04b29="" class="d-flex flex-column w-100">
  <div data-v-8bf04b29="">
    <!-- Desktop Navbar -->
    <v-app-bar app="" color="secondary" dense="" flat="" class="py-2 mb-2 hidden-sm-and-down fill-width">
      <v-container>
        <v-row class="d-flex align-center justify-space-evenly">
          <v-col cols="1">
            <v-img src="../../favicon.ico" alt="Platform Logo" height="40"></v-img>
          </v-col>
          <v-col cols="3">
            <v-text-field modelvalue="" color="primary" bg-color="complementary" placeholder="Pesquisar eventos..." prepend-inner-icon="mdi-magnify" variant="outlined" flat="" hide-details="" rounded=""></v-text-field>
          </v-col>
          <v-col cols="3">
            <v-select items="Localização 1,Localização 2,Localização 3" item-text="name" item-value="id" label="Localizações" prepend-inner-icon="mdi-map-marker" variant="solo" density="comfortable" bg-color="transparent" flat="" rounded="" hide-details="" class="pr-6"></v-select>
          </v-col>
          <v-col cols="4" class="d-flex align-center justify-space-around">
            <v-btn to="[object Object]">Crie seu evento</v-btn>
            <v-btn>Acesse sua conta</v-btn>
            <v-btn class="text-none" color="primary" variant="flat" style="border: 1px solid white;" rounded="xs"><b>Cadastre-se</b></v-btn>
          </v-col>
        </v-row>
      </v-container>
    </v-app-bar><!-- Mobile Navbar -->
    <v-app-bar app="" color="secondary" dense="" flat="" class="py-2 mb-0 hidden-md-and-up" style="position: fixed; top: 0px; left: 0px; right: 0px;">
      <v-container>
        <v-row class="d-flex align-center justify-space-between">
          <v-col cols="2">
            <v-img src="../../favicon.ico" alt="Platform Logo" height="40"></v-img>
          </v-col>
          <v-col cols="2" class="d-flex align-center justify-end">
            <v-btn icon="">
              <v-icon>mdi-menu</v-icon>
            </v-btn>
          </v-col>
        </v-row>
      </v-container>
    </v-app-bar><!-- Mobile Navigation Dialog -->
    <v-dialog modelvalue="false" fullscreen="" hide-overlay="" transition="dialog-bottom-transition">
      <v-card>
        <v-toolbar flat="" color="secondary">
          <v-toolbar-title>Mobile Menu</v-toolbar-title>
          <v-spacer></v-spacer>
          <v-btn icon="">
            <v-icon>mdi-close</v-icon>
          </v-btn>
        </v-toolbar>
        <v-list>
          <v-list-item>
            <v-list-item-title>Pesquisar eventos</v-list-item-title>
          </v-list-item>
          <v-list-item>
            <v-list-item-title>Crie seu evento</v-list-item-title>
          </v-list-item>
          <v-list-item>
            <v-list-item-title>Acesse sua conta</v-list-item-title>
          </v-list-item>
          <v-list-item>
            <v-list-item-title>Cadastre-se</v-list-item-title>
          </v-list-item>
          <v-list-group>
            <v-list-item>
              <v-list-item-title>Localização 1</v-list-item-title>
            </v-list-item>
            <v-list-item>
              <v-list-item-title>Localização 2</v-list-item-title>
            </v-list-item>
            <v-list-item>
              <v-list-item-title>Localização 3</v-list-item-title>
            </v-list-item>
          </v-list-group>
        </v-list>
        <!--v-if-->
      </v-card>
    </v-dialog>
  </div>
  <v-main data-v-8bf04b29="" class="flex-grow-1 pa-16 mb-10">
    <v-container data-v-8bf04b29="" fluid="" class="pa-6 d-flex flex-column align-center">
      <h1 data-v-8bf04b29="" class="form-title">Crie seu Evento</h1>
      <v-form data-v-8bf04b29="" class="pa-3 d-flex flex-column align-center max-width-form">
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Nome do Evento" required="" class="field-size"></v-text-field>
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Data de Início do Evento" type="date" required="" class="field-size"></v-text-field>
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Data de Fim do Evento" type="date" required="" class="field-size"></v-text-field>
        <v-select data-v-8bf04b29="" class="field-size" label="Localização do Evento" modelvalue="" items="" item-title="local_name" item-value="id" required=""></v-select>
        <v-select data-v-8bf04b29="" modelvalue="" items="conference,workshop,seminar,meetup" label="Tipo do Evento" required="" class="field-size"></v-select>
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Quantidade Mínima de Participantes" type="number" required="" class="field-size" min="1"></v-text-field>
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Quantidade Máxima de Participantes" type="number" required="" class="field-size" min="1"></v-text-field>
        <v-text-field data-v-8bf04b29="" modelvalue="" label="Quantidade de Horas" type="number" required="" class="field-size" min="1"></v-text-field>
        <v-textarea data-v-8bf04b29="" modelvalue="" label="Descrição do Evento" required="" class="field-size"></v-textarea>
        <v-row data-v-8bf04b29="" class="mt-4">
          <v-col data-v-8bf04b29="" cols="6" class="d-flex justify-start">
            <v-btn data-v-8bf04b29="" color="secondary" class="btn-size btn-style voltar-btn" elevation="2">Voltar</v-btn>
          </v-col>
          <v-col data-v-8bf04b29="" cols="6" class="d-flex justify-end">
            <v-btn data-v-8bf04b29="" type="submit" color="primary" class="btn-size btn-style" elevation="2">Criar Evento</v-btn>
          </v-col>
        </v-row>
      </v-form>
    </v-container>
  </v-main>
  <v-footer data-v-4896eafd="" data-v-8bf04b29="" color="secondary" class="py-6" width="auto" height="auto">
    <v-container data-v-4896eafd="">
      <v-row data-v-4896eafd="" class="footer-row" align="center" justify="space-between">
        <v-col data-v-4896eafd="" cols="12" md="2" class="d-flex align-center justify-center mb-4 mb-md-0">
          <v-img data-v-4896eafd="" src="../../favicon.ico" alt="Platform Logo" height="40"></v-img>
        </v-col>
        <v-col data-v-4896eafd="" cols="12" md="3" class="text-center text-md-left">
          <h3 data-v-4896eafd="" class="text-h6 font-weight-bold">Encontre Eventos</h3>
          <v-list data-v-4896eafd="" dense="">
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Hoje</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Amanhã</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Esta Semana</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Próxima Semana</v-list-item-title>
            </v-list-item>
          </v-list>
        </v-col>
        <v-col data-v-4896eafd="" cols="12" md="3" class="text-center text-md-left">
          <h3 data-v-4896eafd="" class="text-h6 font-weight-bold">Mapa do site</h3>
          <v-list data-v-4896eafd="" dense="">
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Home</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Eventos</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Login</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Contato</v-list-item-title>
            </v-list-item>
          </v-list>
        </v-col>
        <v-col data-v-4896eafd="" cols="12" md="4" class="text-center text-md-left">
          <h3 data-v-4896eafd="" class="text-h6 font-weight-bold">Contato</h3>
          <v-list data-v-4896eafd="" dense="">
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Email: fakeemail@gmail.com</v-list-item-title>
            </v-list-item>
            <v-list-item data-v-4896eafd="">
              <v-list-item-title data-v-4896eafd="">Celular: 213214124124</v-list-item-title>
            </v-list-item>
          </v-list>
        </v-col>
      </v-row>
    </v-container>
  </v-footer>
  <v-snackbar data-v-8bf04b29="" modelvalue="true" color="red" timeout="3000" top="">Erro ao buscar locais</v-snackbar>
</v-app>"
`;
